pkg/morph/client/notifications.go

```diff
@@ -120,22 +120,23 @@ func (c *Client) ReceiveNotaryRequests(txSigner util.Uint160) error {
 		return ErrConnectionLost
 	}
 
-	c.subs.Lock()
-	defer c.subs.Unlock()
```

Comment:
It's very recent code, likely not the cause of #2559.

Reply:
yeap, that is why i do not say it closes smth. but that is the reason (and the only one) for the recent deadlock that I have found (the info was not full since I was a little late)